src/m365/teams/commands/conversationmember/conversationmember-add.ts

```diff
@@ -0,0 +1,282 @@
+      return 'Specify userId or userDisplayName, one is required';
+    }
+
+    return true;
+  }
+
+  private getTeamId(args: CommandArgs): Promise<string> {
+    if (args.options.teamId) {
+      return Promise.resolve(args.options.teamId);
+    }
+
+    return new Promise<string>((resolve: (channelId: string) => void, reject: (error: any) => void): void => {
+      const teamRequestOptions: any = {
+        url: `${this.resource}/beta/groups?$filter=resourceProvisioningOptions/Any(x:x eq 'Team') and displayName eq '${encodeURIComponent(args.options.teamName as string)}'`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      request
+        .get<{ value: Team[] }>(teamRequestOptions)
+        .then(response => {
+          const teamItem: Team | undefined = response.value[0];
+
+          if (!teamItem) {
+            return reject(`The specified team '${args.options.teamName}' does not exist in Microsoft Teams`);
+          }
+
+          if (response.value.length > 1) {
+            return reject(`Multiple Microsoft Teams with name '${args.options.teamName}' found. Please disambiguate:${os.EOL}${response.value.map(x => `- ${x.id}`).join(os.EOL)}`);
+          }
+
+          return resolve(teamItem.id);
+        }, err => { reject(err) });
+    })
+  }
+
+  private getChannelId(teamId: string, args: CommandArgs): Promise<string> {
+    if (args.options.channelId) {
+      const channelIdRequestOptions: any = {
+        url: `${this.resource}/v1.0/teams/${encodeURIComponent(teamId)}/channels/${encodeURIComponent(args.options.channelId as string)}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      return new Promise<string>((resolve: (channelId: string) => void, reject: (error: any) => void): void => {
+        request
+          .get<Channel>(channelIdRequestOptions)
+          .then((response: Channel) => {
+            const channelItem: Channel | undefined = response;
+            return resolve(channelItem.id);
+          }, (err: any) => {
+            if (err.error && err.error.code == "NotFound") {
+              return reject(`The specified channel '${args.options.channelId}' does not exist or is invalid in the Microsoft Teams team with ID '${teamId}'`);
+            } else {
+              return reject(err);
+            }
+          });
+      });
+    } else {
+      const channelRequestOptions: any = {
+        url: `${this.resource}/v1.0/teams/${encodeURIComponent(teamId)}/channels?$filter=displayName eq '${encodeURIComponent(args.options.channelName as string)}'`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      return new Promise<string>((resolve: (channelId: string) => void, reject: (error: any) => void): void => {
```

Comment:
Same as earlier: let's return the promise chain and avoid wrapping the whole chain in a separate promise. That will help us keep the code more readable.